os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -185,41 +190,39 @@ static void up_stackdump(uint32_t sp, uint32_t stack_base)
 
 static void arm_registerdump(volatile uint32_t *regs)
 {
-	/* Dump the interrupt registers */
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Register dump\n");
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("R0: %08x R1: %08x R2: %08x R3: %08x\n",
-	regs[REG_R0], regs[REG_R1], regs[REG_R2], regs[REG_R3]);
+	if (regs) {
+		/* Dump the interrupt registers */
+		lldbg_noarg("===========================================================Register_Dump===========================================\n");
+		lldbg("R0: %08x %08x %08x %08x",
+		regs[REG_R0], regs[REG_R1], regs[REG_R2], regs[REG_R3]);
 #ifdef CONFIG_ARM_THUMB
-	lldbg_noarg("R4: %08x R5: %08x R6: %08x FP: %08x\n",
-	regs[REG_R4], regs[REG_R5], regs[REG_R6], regs[REG_R7]);
-	lldbg_noarg("R8: %08x SB: %08x SL: %08x R11: %08x\n",
-	regs[REG_R8], regs[REG_R9], regs[REG_R10], regs[REG_R11]);
+		lldbg_noarg(" %08x %08x %08x %08x\n",
+		regs[REG_R4], regs[REG_R5], regs[REG_R6], regs[REG_R7]);
+		lldbg("R8: %08x %08x %08x %08x",
```

Comment:
Could you let me know why you change this?

Reply:
To make it consistent with armv8-m, so that trap tool can also work for armv7-a.